ctapipe/reco/HillasReconstructor.py

```diff
@@ -99,13 +99,67 @@ class that reconstructs the direction of an atmospheric shower
 
     """
 
-    def __init__(self, config=None, parent=None, **kwargs):
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
         super().__init__(config=config, parent=parent, **kwargs)
-        self.hillas_planes = {}
-        self.divergent_mode = False
-        self.corrected_angle_dict = {}
+        self.subarray = subarray
+
+        self.cam_R_m = {
+            tel_id: tel.camera.geometry.guess_radius()
+            for tel_id, tel in subarray.tel.items()
+        }
+
+        telframe = TelescopeFrame()
+        self.cam_R_deg = {
+            tel_id: tel.camera.geometry.transform_to(telframe).guess_radius()
+            for tel_id, tel in subarray.tel.items()
+        }
```

Comment:
Right now `HillasReconstructor.__init__()` takes **over a minute** on my relatively fast machine. I think it is due to this loop, which does a lot of transforms just to get the radius in degrees.  There must be a better way... Two options come to mind:

1. keep doing it this way, but only transform once per camera *type* rather than camera *id*
2. just to transform the radius as a SkyCoord as defined by point along the Y-axis of the camera to the TelescopeFrame and then the new coordinate's delta_alt should be in degrees. 

I think the best is to do option 1 for the first loop (getting the radii in meters), and then use option 2 for the second "degree" loop:

Again, since the cameras are all of similar types and we don't care about pointing, it would be better to store this by camera type rather than id, so the first loop would be:

```py3
self._cam_radius_m = {
    cam.camera_name, cam.geometry.guess_radius()
    for cam in subarray.camera_types  # note now it's by type
}
```
which looks like
```
{'ASTRICam': <Quantity 0.19680761 m>,
 'FlashCam': <Quantity 1.08007852 m>,
 'LSTCam': <Quantity 1.10719268 m>}
```
later  you have to look them up by camera name

Then, he degree loop should be something like:

```py3
for tel_id, radius_m in self.cam_R_m.items():
    point_cam = SkyCoord(0, radius_m, self.subarray.tel[tel_id].camera.geometry.frame)
    point_tel = point_m.transform_to(telframe)
    self._cam_radius_deg[tel_id] = point_tel.delta_alt
```

Reply:
Please don't use `camera.camera_name`, just use the camera itself, this is the reason why we made them hashable.

Reply:
you could also use the CameraDescription itself as the key rather than the camera name string, but not sure if that is faster to compare later or not (depends on how fast `CameraDescription.__hash__()` is).

Reply:
This is what I showed, isn't it?

Reply:
I tried with:

```python
self._cam_radius_m = {
            cam: cam.geometry.guess_radius()
            for cam in subarray.camera_types
        }

camera = self.subarray.tel[tel_id].camera
self._cam_radius_m[camera]
```

also tried with this `hash()` method, but none of the 2 ways works...
Honestly, I found the `camera_name` method much easier and useful also for debugging, but maybe I am missing something

Reply:
> This is what I showed, isn't it?

I think we wrote those comments simultaneously

Reply:
The code you pasted above works, except for the last line,  which should be:
```py3
 self._camera_radus_m[camera.geometry]
```
since the keys you used were `CameraGeometries` not `CameraDescriptions` (that is a confusion really with SubarrayDescription.camera_types, which probably didn't get updated when CameraDescriptions were introduced, but we can fix that later)

Reply:
Also, it's true that we should just make that a lazyproperty in CameraGeometry, but let's leave that for another PR

Reply:
camera name must not be used, it is a guessed and only available for known cameras. We must nlt rely on it, and it can oly be used for printing / labeling, not for any kind of actual analysis.

You sa, you tried but did not say that it didnt work. So what is the problem woth the  code you showed?